### PR TITLE
Specs

### PR DESCRIPTION
**Pull request by shanecav84**
I've started adding specs. How's this looking so far?

### CODE REVIEW

**Comment by shanecav84** [comment]
@jgarber623 How's this spec setup work for you? I'm writing specs and force-pushing as I go. I'd like to get these specs done then get FrancisCMS to support Rails 5.

**Comment by jgarber623** [comment]
Awesome! Thanks for the PR, @shanecav84! 💪 

Apologies for the delay in posting here. Taking a look now commit-by-commit and will post comments/questions as a review.

**Comment by jgarber623** [comment]
Also, it may be worthwhile adding `spec/dummy/**/*` to `.rubocop.yml`:

```yml
AllCops:
  Exclude:
    - 'db/**/*'
    - 'spec/dummy/**/*'
    - 'vendor/**/*'
```

**Comment by jgarber623** [comment]
Also also: Adding some details on how to run the test suite to `README.md` (in the "Testing" section) would be super helpful, too.

File: spec/factories/links.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :link, class: ::FrancisCms::Link do
+    url { ::Faker::Internet.url }
+    title { ::SecureRandom.hex }
+    body { ::Faker::Markdown.random }
```
[the remark this answers is not in the document]
Comment (shanecav84) [reply spec/factories/links.rb]: Nope, not necessary. I generally add the leading `::` for short, top-level namespaces. How do you prefer it?

File: app/models/francis_cms/link.rb [hunk 2]
```diff
@@ -40,6 +40,12 @@ def youtube?
       parsed_url.host.gsub(/^www\./, '').match(/^youtube\.com/)
     end
 
+    def parsed_query
+      return if parsed_url.query.blank?
+
+      CGI.parse(parsed_url.query)
+    end
```
[the remark this answers is not in the document]
Comment (shanecav84) [reply app/models/francis_cms/link.rb]: YouTube offers the form `https://youtu.be/dQw4w9WgXcQ` for linking to YouTube videos. In that case, `parsed_url.query` is `nil` which raises an error when passed to `CGI.parse`.

I need to take a closer look at this to take into account this shortened form.

[the remark this answers is not in the document]
Comment (shanecav84) [reply app/models/francis_cms/link.rb]: I've updated the `Link` model to support `youtu.be` links.

File: spec/factories/links.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgarber623) [reply spec/factories/links.rb]: I tend to go without when they aren't necessary, but I can see the benefit of including them as a guard against conflicting module names and as a visible flag to other developers.

**Comment by shanecav84** [comment]
- Added instructions to README
- `spec/` passes rubocop

I'm reaching the point where I don't understand the implementation enough to come up with adequate tests; specifically webmentions and syndications.

**Pull request closed without merging**